add recipe for inspire

Simon-Lin opened the pull request:
### Brief summary of what the package does

inspire is an Emacs interface for literature and references searching on the high energy article database [inspirehep](https://inspirehep.net/).

### Direct link to the package repository

https://github.com/Simon-Lin/inspire.el

### Your association with the package

I am the author.

### Relevant communications with the upstream package maintainer

 **None needed**

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

Some false-positives:

- with `package-lint`:
596:3: error: `authors` was removed in Emacs version 25.1.
987:16: error: `authors` was removed in Emacs version 25.1.
1038:10: error: `authors` was removed in Emacs version 25.1.
`authors` is a custom variable name used here.

- with `byte-compile`:
inspire.el:881:6: Warning: `inspire--insert-with-face` called with 1 argument, but requires 2+
I am not sure why I got this warning but the function is indeed called with 2 arguments, so I assume this is some bug in the compiler?

riscy commented:
Thanks for this.  You already noted some of these as false positives, but I'll re-paste some here and go into depth:

`inspire.el` with byte-compile using Emacs 28.1:
```
inspire.el:881:6: Warning: inspire--insert-with-face called with 1 argument,
    but requires 2+
In end of data:
inspire.el:611:10: Warning: the function `url-insert' is not known to be
    defined.
```
It's called with one argument on line 844, `(inspire--insert-with-face (alist-get 'name author))`; sometimes the byte-compiler is a little off (sometimes it's a *lot* off when macros are involved).  The second warning looks like a `(require 'url-handlers)` is required.

`inspire.el` with package-lint 20221003.1636:
```
3 issues found:
596:3: error: `authors' was removed in Emacs version 25.1.
987:16: error: `authors' was removed in Emacs version 25.1.
1038:10: error: `authors' was removed in Emacs version 25.1.
```
The way I recommend getting around this is to use `authors_` instead; the extra `_` is a very small price to pay for a 100% clean package-lint where you don't get used to ignoring errors.

`inspire.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- inspire.el#L757: It's safer to sharp-quote function names; use `#'` (`apply #'propertize ...`)
- inspire.el#L358: It's safer to sharp-quote function names; use `#'`
- inspire.el#L809: It's safer to sharp-quote function names; use `#'`
- inspire.el#L811: It's safer to sharp-quote function names; use `#'`
- inspire.el#L847: It's safer to sharp-quote function names; use `#'`
- inspire.el#L858: It's safer to sharp-quote function names; use `#'`
- inspire.el#L860: It's safer to sharp-quote function names; use `#'`
- inspire.el#L1090: It's safer to sharp-quote function names; use `#'`
- inspire.el#L1129: It's safer to sharp-quote function names; use `#'`
- inspire.el#L676: It's safer to sharp-quote function names; use `#'`
- inspire.el#L983: No `format` required; `error` takes an f-string
- inspire.el#L1072: No `format` required; `error` takes an f-string
```

Simon-Lin commented:
Thanks. I have addressed the issues based on your suggestion. See https://github.com/Simon-Lin/inspire.el/commit/1157f8e16b0e7b6d68184993cb94332b0cb5494c.

riscy commented:
Thanks again, this looks like it will be a useful package - and the checklist itself looks great now.  A couple more comments from a more careful pass:

The package name "inspire" is fairly generic and a common word; I think "inspirehep" as a package name would both mirror the URL that people are familiar with, and serve as disambiguation.  Not a blocker, but let me know your thoughts

In your summary line "Emacs interface for inspirehep.net", you could drop the "Emacs"; Emacs is not a salient keyword in this context.

Very minor, but these backslashes ("\[") are not having any effect and might as well be removed (these occur in three places):
```elisp
    (define-key map (kbd "\[") #'inspire-previous-search)
    (define-key map (kbd "\]") #'inspire-next-search)
```

This is minor, but downstream facilities to display your `;;; Commentary` typically don't know how to word wrap it; you should do this yourself using e.g. M-q to improve the user experience

Consider using `let*` instead of `let` when appropriate.  For example a form like this:
```elisp
  (let ((metadata (alist-get 'metadata entry))
	(ids) (inspire-bai) (orcid) (control-number) (name) (native-names) (advisors) (positions) (current-position) (arxiv-categories) (projects) (email) (timestamp))
    (setq ids (alist-get 'ids metadata))
    (setq inspire-bai (alist-get 'value (seq-find (lambda (x) (equal (alist-get 'schema x) "INSPIRE BAI")) ids)))
```
I'd start writing like this:
```elisp
  (let* ((metadata (alist-get 'metadata entry))
	 (ids (alist-get 'ids metadata))
         (inspire-bai (alist-get 'value (seq-find (lambda (x) (equal (alist-get 'schema x) "INSPIRE BAI")) ids)))
```

Altogether these are minor; let me know what you think about the name, and this can be an easy merge.

Simon-Lin commented:
Thanks again for the review and recommendation.
The issue with the package name is that there is an existing package called [inspirehep.el](https://codeberg.org/rahguzar/inspirehep.el) that does a very similar thing (except it is based on [biblo.el](https://github.com/cpitclaudel/biblio.el) while mine is mostly based on [arxiv-mode](https://github.com/fizban007/arxiv-mode)). The mentioned package is not on melpa, but I would like to avoid a possible name clash. I agree that "inspire" may be too generic, but I don't have a better alternative.

riscy commented:
I think this is fine in this case -- and thanks for being conscientious and watching out for collisions with other packages' names.
